### PR TITLE
uv 0.1.0 (new formula)

### PR DESCRIPTION
**Pull request by baodrate**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Might be a big premature, but adding the project from the creators of ruff. Announcement post: https://astral.sh/blog/uv

### CODE REVIEW

File: Formula/u/uv.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class Uv < Formula
+  desc "Extremely fast Python package installer and resolver, written in Rust"
+  homepage "https://github.com/astral-sh/uv"
```
Comment (chenrui333) [inline Formula/u/uv.rb:3]:
```suggestion
  homepage "https://astral.sh/"
```

File: Formula/u/uv.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  url "https://github.com/astral-sh/uv/archive/refs/tags/0.1.0.tar.gz"
+  sha256 "4c0748088b87672383a796d9ad198b7ead74b636afd024ed21872b4a6fd7cadd"
+  license any_of: ["MIT", "Apache-2.0"]
+  head "https://github.com/astral-sh/uv.git", branch: "main"
+
+  depends_on "rust" => :build
```
Comment (chenrui333) [inline Formula/u/uv.rb:9]:
```suggestion
  depends_on "pkg-config" => :build
  depends_on "rust" => :build
  depends_on "openssl@3"
```

File: Formula/u/uv.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  def install
+    system "cargo", "install", "--no-default-features", *std_cargo_args(path: "crates/uv")
```
Comment (chenrui333) [inline Formula/u/uv.rb:12]:
```suggestion
    # Ensure that the `openssl` crate picks up the intended library.
    ENV["OPENSSL_DIR"] = Formula["openssl@3"].opt_prefix
    ENV["OPENSSL_NO_VENDOR"] = "1"

    system "cargo", "install", "--no-default-features", *std_cargo_args(path: "crates/uv")
```

File: Formula/u/uv.rb (hunk 1 above)
Comment (baod-rate) [reply Formula/u/uv.rb:3]: https://astral.sh actually does not feature the project at all (except for the aforementioned blog post). Wouldn't it be confusing to be directed to a site that doesn't mention the project?

Comment (chenrui333) [reply Formula/u/uv.rb:3]: then should this get changed?

image

File: Formula/u/uv.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  desc "Extremely fast Python package installer and resolver, written in Rust"
+  homepage "https://github.com/astral-sh/uv"
+  url "https://github.com/astral-sh/uv/archive/refs/tags/0.1.0.tar.gz"
+  sha256 "4c0748088b87672383a796d9ad198b7ead74b636afd024ed21872b4a6fd7cadd"
+  license any_of: ["MIT", "Apache-2.0"]
+  head "https://github.com/astral-sh/uv.git", branch: "main"
+
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
+  depends_on "openssl@3"
```
Comment (chenrui333) [inline Formula/u/uv.rb:11]:
```suggestion
  depends_on "pkg-config" => :build
  depends_on "rust" => :build
  depends_on "libgit2"
  depends_on "openssl@3"
```

File: Formula/u/uv.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+  def install
+    # Ensure that the `openssl` crate picks up the intended library.
```
Comment (chenrui333) [inline Formula/u/uv.rb:14]:
```suggestion
    ENV["LIBGIT2_NO_VENDOR"] = "1"

    # Ensure that the `openssl` crate picks up the intended library.
```

**Comment by chenrui333** [comment]
Looks like you also need `libgit2` dependency, and be sure to squash all the commits for `commit code style`

File: Formula/u/uv.rb (hunk 1 above)
Comment (p-linnane) [reply Formula/u/uv.rb:3]: Given that there is nothing on the homepage yet, I think it's fine to just use the repo for now.

**Review by p-linnane — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@baod-rate, thanks for your first contribution to homebrew-core! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**